dbms/src/Core/BackgroundSchedulePool.h

```diff
@@ -1,156 +1 @@
-        friend class TaskNotification;
-        friend class BackgroundSchedulePool;
-
-        void execute();
-
-        void scheduleImpl(std::lock_guard<std::mutex> & schedule_mutex_lock);
-
-        BackgroundSchedulePool & pool;
-        std::string log_name;
-        TaskFunc function;
-
-        std::mutex exec_mutex;
-        std::mutex schedule_mutex;
-
-        /// Invariants:
-        /// * If deactivated is true then scheduled, delayed and executing are all false.
-        /// * scheduled and delayed cannot be true at the same time.
-        bool deactivated = false;
-        bool scheduled = false;
-        bool delayed = false;
-        bool executing = false;
-
-        /// If the task is scheduled with delay, points to element of delayed_tasks.
-        DelayedTasks::iterator iterator;
-    };
-
-    class TaskHolder
-    {
-    public:
-        TaskHolder() = default;
-        explicit TaskHolder(const TaskInfoPtr & task_info_) : task_info(task_info_) {}
-        TaskHolder(const TaskHolder & other) = delete;
-        TaskHolder(TaskHolder && other) noexcept = default;
-        TaskHolder & operator=(const TaskHolder & other) noexcept = delete;
-        TaskHolder & operator=(TaskHolder && other) noexcept = default;
-
-        ~TaskHolder()
-        {
-            if (task_info)
-                task_info->deactivate();
-        }
-
-        TaskInfo * operator->() { return task_info.get(); }
-        const TaskInfo * operator->() const { return task_info.get(); }
-
-    private:
-        TaskInfoPtr task_info;
-    };
-
-    TaskHolder createTask(const std::string & log_name, const TaskFunc & function);
-
-    size_t getNumberOfThreads() const { return size; }
-
-    BackgroundSchedulePool(size_t size);
-    ~BackgroundSchedulePool();
-
-private:
-    using Threads = std::vector<std::thread>;
-
-    void threadFunction();
-    void delayExecutionThreadFunction();
-
-    /// Schedule task for execution after specified delay from now.
-    void scheduleDelayedTask(const TaskInfoPtr & task_info, size_t ms, std::lock_guard<std::mutex> & task_schedule_mutex_lock);
-
-    /// Remove task, that was scheduled with delay, from schedule.
-    void cancelDelayedTask(const TaskInfoPtr & task_info, std::lock_guard<std::mutex> & task_schedule_mutex_lock);
-
-    /// Number for worker threads.
-    const size_t size;
-    std::atomic<bool> shutdown {false};
-    Threads threads;
-    Poco::NotificationQueue queue;
-
-    /// Delayed notifications.
-
-    std::condition_variable wakeup_cond;
-    std::mutex delayed_tasks_mutex;
-    /// Thread waiting for next delayed task.
-    std::thread delayed_thread;
-    /// Tasks ordered by scheduled time.
-    DelayedTasks delayed_tasks;
-
-    /// Thread group used for profiling purposes
-    ThreadGroupStatusPtr thread_group;
-
-    void attachToThreadGroup();
-};
-
-using BackgroundSchedulePoolPtr = std::shared_ptr<BackgroundSchedulePool>;
-
-}
+#include <Misc/BackgroundSchedulePool.h>
```

Comment:
Very dirty hack. Why we cannot use %<Misc/BackgroundSchedulePool.h>% everywhere?

Reply:
Its temporary hack, i will make mass replace to Misc/BackgroundSchedulePool.h if anything other is ok